### ISSUE TITLE
Error when you delete all steps (Trac #100)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#104]
An error occurs if the user deletes all of the protocol steps. The user should be prevented from deleting the final step.

C:\Python27\lib\site-packages\pygtkhelpers\ui\objectlist\view.py:358: GtkWarning: gtk_tree_path_append_index: assertion `index >= 0' failed
  return self.model[iter][0]
C:\Python27\lib\site-packages\pygtkhelpers\ui\objectlist\view.py:358: GtkWarning: gtk_tree_model_get_iter: assertion `path->depth > 0' failed
  return self.model[iter][0]
ERROR: microdrop.gui.protocol_grid_controller plugin crashed processing on_step_removed signal.
Reason: could not find tree path
ERROR: microdrop.gui.protocol_grid_controller plugin crashed processing on_step_created signal.
Reason: type object argument after ** must be a mapping, not NoneType
ERROR: microdrop.gui.protocol_grid_controller plugin crashed processing on_step_swapped signal.
Reason: could not find tree path

Migrated from http://microfluidics.utoronto.ca/microdrop/ticket/100
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "An error occurs if the user deletes all of the protocol steps. The user should be prevented from deleting the final step.\n\nC:\\Python27\\lib\\site-packages\\pygtkhelpers\\ui\\objectlist\\view.py:358: GtkWarning: gtk_tree_path_append_index: assertion `index >= 0' failed\n  return self.model[iter][0]\nC:\\Python27\\lib\\site-packages\\pygtkhelpers\\ui\\objectlist\\view.py:358: GtkWarning: gtk_tree_model_get_iter: assertion `path->depth > 0' failed\n  return self.model[iter][0]\nERROR: microdrop.gui.protocol_grid_controller plugin crashed processing on_step_removed signal.\nReason: could not find tree path\nERROR: microdrop.gui.protocol_grid_controller plugin crashed processing on_step_created signal.\nReason: type object argument after ** must be a mapping, not NoneType\nERROR: microdrop.gui.protocol_grid_controller plugin crashed processing on_step_swapped signal.\nReason: could not find tree path", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "Error when you delete all steps", 
    "priority": "major", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2012-04-17T19:03:52", 
    "milestone": "Microdrop 1.0", 
    "owner": "cfobel", 
    "type": "defect"
}
```

**Comment by cfobel** [comment]
Trac update at 20120417T23:20:20: 
* christian@fobel.net commented:

> commit 0d0c1074306e6033934f4632bf5801d8cdcadf55
> Author: Christian Fobel <christian@fobel.net>
> Date:   Tue Apr 17 19:11:27 2012 -0400
> 
>     Move insert/delete from buttons to popup menu
>     
>     Fixes #105.  Note that this also fixes #103 and fixes #104 after updating the
>     dmf_control_board plugin to revision v0.1-216-g03278e8.


* christian@fobel.net changed resolution from "" to "fixed"
* christian@fobel.net changed status from "new" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "" to "microdrop"
